pandas/io/pytables.py

```diff
@@ -3100,17 +3099,23 @@ def read(self, start=None, stop=None, **kwargs):
             axes.append(ax)
 
         items = axes[0]
-        blocks = []
+        dfs = []
+
```

Comment:
i think it would be better to use a list comprehension here and make a function for the DataFrame creation (e.g. lines 3106-3110), but can be a followup